src/main/java/teammates/storage/api/StudentsDb.java

```diff
@@ -306,11 +395,41 @@ public StudentAttributes getStudentForRegistrationKey(String registrationKey) {
      */
     @Deprecated
     public List<StudentAttributes> getAllStudents() {
+        Map<String, StudentAttributes> result = new LinkedHashMap<String, StudentAttributes>();
+        
+        for (StudentAttributes student : getAllOldStudents()) {
+            result.put(student.getId(), student);
+        }
+        for (StudentAttributes student : getAllCourseStudents()) {
+            result.put(student.getId(), student);
+        }
+        return new ArrayList<>(result.values());
+    }
+
+    @Deprecated
+    public List<StudentAttributes> getAllOldStudents() {
+        List<StudentAttributes> students = new ArrayList<>();
+        List<Student> studentEntities = getStudentEntities();
+        
+        for (Student student : studentEntities) {
+            if (!JDOHelper.isDeleted(student)) {
+                students.add(new StudentAttributes(student));
+            }
+        }
+        return students;
+    }
+    
+    /**
+     * This method is not scalable. Not to be used unless for admin features.
+     * @return the list of all students in the database.
+     */
+    @Deprecated
+    public List<StudentAttributes> getAllCourseStudents() {
         List<StudentAttributes> list = new LinkedList<StudentAttributes>();
-        List<Student> entities = getStudentEntities();
-        Iterator<Student> it = entities.iterator();
+        List<CourseStudent> entities = getCourseStudentEntities();
+        Iterator<CourseStudent> it = entities.iterator();
         while (it.hasNext()) {
```

Comment:
Even though it was like this before, do we have reasons to keep it as iterator instead of using for-loop?